tests/cpp/tree/test_quantile_hist.cc

```diff
@@ -622,18 +580,6 @@ TEST(QuantileHist, InitData) {
   maker_float.TestInitData();
 }
 
-TEST(QuantileHist, InitDataSampling) {
```

Comment:
I am removing the unit test `QuantileHist.InitDataSampling`, since we no longer have the guarantee that using the same seed will lead to the same set of sampled rows. Changing the number of threads will now lead to a different sample, even when `seed` is fixed.

Reply:
@RAMitchell @trivialfis Can we live with the relaxed guarantee of reproduciblity? Now the user needs to fix `seed` as well as `nthread` to obtain fully reproducible results.

Reply:
In this case I don't think we should merge it, the reproducibility guarantee is important. Ideally we would have something like this https://github.com/rabauke/trng4/blob/master/examples/pi_block_openmp.cc, but trng seems way to heavy of a dependency for this purpose.

Reply:
@RAMitchell Got it, let's review the alternative #6410 then, which does not break reproducibility.